src/tableediting.js

```diff
@@ -214,16 +214,25 @@ export default class TableEditing extends Plugin {
 			}
 
 			const isLastCellInRow = currentCellIndex === tableRow.childCount - 1;
-			const isLastRow = currentRowIndex === table.childCount - 1;
+			let isLastRow = currentRowIndex === table.childCount - 1;
 
 			if ( isForward && isLastRow && isLastCellInRow ) {
-				editor.plugins.get( 'TableUtils' ).insertRows( table, { at: table.childCount } );
+				editor.execute( 'insertTableRowBelow' );
+
+				// Re-evaluate `isLastRow`. If `insertTableRowBelow` execution didn't add any row (because it was disabled or it got
+				// overwritten in some way) this will still be `true`. But if the row was added it will change to `false`.
+				isLastRow = currentRowIndex === table.childCount - 1;
 			}
 
 			let cellToFocus;
 
 			// Move to first cell in next row.
 			if ( isForward && isLastCellInRow ) {
+				if ( isLastRow ) {
```

Comment:
I don't like the `if` in `if` here. I'd move that check as `if( !command.isEnabled ) { return }` to the previous check of inserting table row.